app/workers/concerns/code_analysis_mixin.rb

```diff
@@ -19,11 +31,64 @@ def merged_linter_results
     results
   end
 
+  # run linters via pronto and return the pronto result
+  def pronto_result
+    p_result = nil
+
+    # temporary solution for: download repo, obtain changes, get pronto result about changes
+    Dir.mktmpdir do |dir|
+      FileUtils.copy_entry(@branch.repo.path.to_s, dir)
+      repo = Pronto::Git::Repository.new(dir)
+      rg = repo.instance_variable_get(:@repo)
+      rg.fetch('origin', @branch.name.sub(/^prs/, 'pull'))
+      rg.checkout('FETCH_HEAD')
+      rg.reset('HEAD', :hard)
+      patches = repo.diff(@branch.merge_target)
```

Comment:
This is already possible with the repos from the repos directory.  Unfortunately, Pronto::Git::Repository seems to only want to use head, but it doesn't look like a stretch to just "copy" what they do there (or make a PR to pronto to support non-head commits, which should be straightforward).  I think you can just do something like

```ruby
branch.repo.git_fetch # note that branch is available to the CodeAnalysisMixin directly via the BranchWorkerMixin as they are usually together
git_service = branch.git_service
merge_base = git_service.merge_base(commit)
diff = git_service.rugged_repo.diff(merge_base, "refs/remotes/origin/#{branch.merge_target}", options)  # Getting the merge target ref is actually already a part of lib/git_service/branch.rb, but it's not exposed in an easy way, so just hardcoding here for an example
patches = Pronto::Git::Patches.new(Pronto::Git::Repository.new(branch.repo.path, merge_base, diff)
p_result = Pronto::Runners.new.run(patches)
```